src/coreclr/nativeaot/Runtime/unix/PalCreateDump.cpp

```diff
@@ -0,0 +1,583 @@
+            exit(-1);
+        }
+    }
+    else
+    {
+#if HAVE_PRCTL_H && HAVE_PR_SET_PTRACER
+        // Gives the child process permission to use /proc/<pid>/mem and ptrace
+        if (prctl(PR_SET_PTRACER, childpid, 0, 0, 0) == -1)
+        {
+            // Ignore any error because on some CentOS and OpenSUSE distros, it isn't
+            // supported but createdump works just fine.
+            fprintf(stderr, "CreateCrashDump: prctl() FAILED %s (%d)\n", strerror(errno), errno);
+        }
+#endif // HAVE_PRCTL_H && HAVE_PR_SET_PTRACER
+        close(child_pipe);
+
+        // Read createdump's stderr messages (if any)
+        if (errorMessageBuffer != nullptr)
+        {
+            // Read createdump's stderr
+            int bytesRead = 0;
+            int count = 0;
+            while ((count = read(parent_pipe, errorMessageBuffer + bytesRead, cbErrorMessageBuffer - bytesRead)) > 0)
+            {
+                bytesRead += count;
+            }
+            errorMessageBuffer[bytesRead] = 0;
+            if (bytesRead > 0)
+            {
+                fputs(errorMessageBuffer, stderr);
+            }
+        }
+        close(parent_pipe);
+
+        // Parent waits until the child process is done
+        int wstatus = 0;
+        int result = waitpid(childpid, &wstatus, 0);
+        if (result != childpid)
+        {
+            fprintf(stderr, "Problem waiting for createdump: waitpid() FAILED result %d wstatus %08x errno %s (%d)\n",
+                result, wstatus, strerror(errno), errno);
+            return false;
+        }
+        else
+        {
+#ifdef _DEBUG
+            fprintf(stderr, "waitpid() returned successfully (wstatus %08x) WEXITSTATUS %x WTERMSIG %x\n", wstatus, WEXITSTATUS(wstatus), WTERMSIG(wstatus));
+#endif
+            return !WIFEXITED(wstatus) || WEXITSTATUS(wstatus) == 0;
+        }
+    }
+    return true;
+}
+
+/*++
+Function:
+  PalCreateCrashDumpIfEnabled
+
+  Creates crash dump of the process (if enabled). Can be called from the unhandled native exception handler.
+
+Parameters:
+    signal - POSIX signal number or 0
+    siginfo - signal info or nullptr
+
+(no return value)
+--*/
+void
+PalCreateCrashDumpIfEnabled(int signal, siginfo_t* siginfo)
+{
+    // If enabled, launch the create minidump utility and wait until it completes
+    if (g_argvCreateDump[0] != nullptr)
+    {
+        const char* argv[MAX_ARGV_ENTRIES];
+        char* signalArg = nullptr;
+        char* crashThreadArg = nullptr;
+        char* signalCodeArg = nullptr;
+        char* signalErrnoArg = nullptr;
+        char* signalAddressArg = nullptr;
+
+        // Copy the createdump argv
+        int argc = 0;
+        for (; argc < MAX_ARGV_ENTRIES; argc++)
+        {
+            argv[argc] = g_argvCreateDump[argc];
+            if (g_argvCreateDump[argc] == nullptr)
+            {
+                break;
+            }
+        }
+
+        if (signal != 0)
```

Comment:
This assumes argc < MAX_ARGV_ENTRIES

Reply:
It should never be even close to MAX_ARGV_ENTRIES here because it is big enough all the options built in BuildCreateDumpCommandLine and all of these options, but I can add a argc < MAX_ARGV_ENTRIES check.